Add schemas for Fusion resource pack metadata

SuperMartijn642 opened the pull request:
[Fusion](https://github.com/SuperMartijn642/Fusion) is a mod for the game Minecraft which allows additional configuration for resource packs for the game. This PR adds two entries to the `catalog.json` file which point to schemas in the master branch of the Fusion repository.

The first entry adds a schema for the properties of the `pack.mcmeta` file added by Fusion. The second entry adds a schema for properties in texture metadata files added by Fusion.
Both entries specifically target `.mcmeta` files, which I believe should be sufficient to not conflict with non-Minecraft files.

This should hopefully help people trying to make resource packs for Fusion to automatically find the correct schemas. 🙂

hyperupcall commented:
thanks for your contribution - it appears that there are `fileMatch` conflicts with the non-fusion resource pack metadata. In particular, `"Minecraft Resource Pack Texture Mcmeta"` and `"Minecraft Data Pack Metadata"`. Is it possible to extend those with fusion-specific properties rather than create new schemas?

SuperMartijn642 commented:
I have added the schemas for the Fusion metadata to the repository, as, from looking at other PRs and issues, it seems I can't reference external schemas.
I adjusted the default Minecraft schemas to include a reference to the Fusion schemas and I have also added some positive tests for the Fusion schemas.

Let me know if I did everything correctly and if this is the correct way to go about it 🙂

hyperupcall commented on src/api/json/catalog.json:
```diff
@@ -1873,6 +1873,16 @@
+    {
+      "name": "Fusion resource pack metadata",
+      "description": "Fusion properties for Minecraft resource pack metadata",
+      "url": "https://json.schemastore.org/fusion-pack-metadata.json"
```
I think it would be good to be consistent and keep the `fileMatch` property, even if it is empty. We have had some past issues with trailing commas breaking some editors, and I just want to be sure

```suggestion
      "fileMatch": [], "url": "https://json.schemastore.org/fusion-pack-metadata.json"
```

hyperupcall commented on src/api/json/catalog.json:
```diff
@@ -1873,6 +1873,16 @@
+    {
+      "name": "Fusion texture metadata",
+      "description": "Fusion properties for Minecraft texture metadata",
+      "url": "https://json.schemastore.org/fusion-texture-metadata.json"
```
```suggestion
      "fileMatch": [], "url": "https://json.schemastore.org/fusion-texture-metadata.json"
```

hyperupcall reviewed (changes requested):
Happy holidays! The new schema looks good to me! Would it be possible that the new schema names be renamed to be prefixed with `partial-` (i.g. `partial-fusion-pack-metadata.json` and `partialfusion-texture-metadata.json`)? I haven't managed to put this in `CONTRIBUTING.md` yet but for schemas that don't match a particular file, we prefix it to make it clear that it's a subschema. Once that is fixed, and the `fileMatch`es are added back like in the review comment, then I will merge this!

SuperMartijn642 commented on src/schemas/json/minecraft-texture-mcmeta.json:
```diff
@@ -2,50 +2,63 @@
-    "animation": {
-      "description": "Animation",
-      "type": "object",
+  "oneOf": [
```
Yup, that is intentional. I have now added a negative test for it as well.

SuperMartijn642 commented:
I have renamed the schemas with the `partial-` prefix and I have also moved the tests to the regular Minecraft schemas which include them (as that is really where they should be).

Let me know if there's any issues of course and have a happy new year!

hyperupcall commented:
thanks! lgtm
